src/index.ts

```diff
@@ -1 +1,326 @@
+} as const;
+/**
+ * A map of versioned Android development tools and their corresponding package names and paths to their binary relative
+ * to `$ANDROID_HOME`.
+ */
+export const versionedAndroidTools = {
+    aapt: { path: (v) => `build-tools/${v}/aapt`, package: 'build-tools' },
+    'split-select': { path: (v) => `build-tools/${v}/split-select`, package: 'build-tools' },
+    zipalign: { path: (v) => `build-tools/${v}/zipalign`, package: 'build-tools' },
+    bcc_compat: { path: (v) => `build-tools/${v}/bcc_compat`, package: 'build-tools' },
+    dexdump: { path: (v) => `build-tools/${v}/dexdump`, package: 'build-tools' },
+    aidl: { path: (v) => `build-tools/${v}/aidl`, package: 'build-tools' },
+    lld: { path: (v) => `build-tools/${v}/lld`, package: 'build-tools' },
+    aapt2: { path: (v) => `build-tools/${v}/aapt2`, package: 'build-tools' },
+    'llvm-rs-cc': { path: (v) => `build-tools/${v}/llvm-rs-cc`, package: 'build-tools' },
+    apksigner: { path: (v) => `build-tools/${v}/apksigner`, package: 'build-tools' },
+    d8: { path: (v) => `build-tools/${v}/d8`, package: 'build-tools' },
+
+    cmake: { path: (v) => `cmake/${v}/bin/cmake`, package: 'cmake' },
+    cpack: { path: (v) => `cmake/${v}/bin/cpack`, package: 'cmake' },
+    ninja: { path: (v) => `cmake/${v}/bin/ninja`, package: 'cmake' },
+    ctest: { path: (v) => `cmake/${v}/bin/ctest`, package: 'cmake' },
+
+    screenshot2: { path: (v) => `cmdline-tools/${v}/bin/screenshot2`, package: 'cmdline-tools' },
+    apkanalyzer: { path: (v) => `cmdline-tools/${v}/bin/apkanalyzer`, package: 'cmdline-tools' },
+    profgen: { path: (v) => `cmdline-tools/${v}/bin/profgen`, package: 'cmdline-tools' },
+    retrace: { path: (v) => `cmdline-tools/${v}/bin/retrace`, package: 'cmdline-tools' },
+    avdmanager: { path: (v) => `cmdline-tools/${v}/bin/avdmanager`, package: 'cmdline-tools' },
+    lint: { path: (v) => `cmdline-tools/${v}/bin/lint`, package: 'cmdline-tools' },
+    sdkmanager: { path: (v) => `cmdline-tools/${v}/bin/sdkmanager`, package: 'cmdline-tools' },
+} as const satisfies Record<string, { path: (version: string) => string; package: string }>;
+/* eslint-enable camelcase */
+
+/** The name of an Android development tool, either unversioned or versioned. */
+export type AndroidToolName = keyof typeof versionedAndroidTools | keyof typeof unversionedAndroidTools;
+/**
+ * An Android development tool, specified as either a name (which will cause the latest installed or available version
+ * to be used) or a name with an explicit version (only possible for versioned tools).
+ *
+ * The version in this context is _not_ the version of the tool itself, but rather the version of the package that
+ * contains the tool.
+ */
+export type AndroidTool = AndroidToolName | { tool: keyof typeof versionedAndroidTools; packageVersion: string };
+
+const platform = process.platform;
+if (platform !== 'win32' && platform !== 'darwin' && platform !== 'linux')
```

Comment:
Ah, I guess Google doesn’t publish the tools for other platforms (though I’d guess BSD works as well). Can you mention this in a comment?

Reply:
Also, maybe you should put this in the beginning of the functions and not in the module code? This would allow users of other platforms to import this module without needing to use dynamic imports.